include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -482,41 +492,487 @@ namespace internal
+            dof_handler_fine, mg_level_fine - 1, [&](const auto &cell) {
+              if (cell->has_children())
+                {
+                  unsigned int c = 0;
+
+                  dof_indices_patch.resize(n_dofs_per_cell_fine);
+
+                  for (const auto &child : cell->child_iterators())
+                    {
+                      dof_indices_cell.resize(
+                        child->get_fe().n_dofs_per_cell());
+                      child->get_mg_dof_indices(dof_indices_cell);
+
+                      for (unsigned int i = 0; i < n_dofs_per_cell_coarse; ++i)
+                        {
+                          const auto index =
+                            dof_indices_cell[lexicographic_numbering[i]];
+
+                          dof_indices_patch[cell_local_children_indices[c][i]] =
+                            index;
+                        }
+
+                      c++;
+                    }
+
+                  indices.insert(indices.end(),
+                                 dof_indices_patch.begin(),
+                                 dof_indices_patch.end());
+                }
+            });
+        }
+
+      std::sort(indices.begin(), indices.end());
+      indices.erase(std::unique(indices.begin(), indices.end()), indices.end());
+      is_locally_relevant_dofs.add_indices(indices.begin(), indices.end());
+    }
+
+    virtual ~FistChildPolicyFineDoFHandlerView() = default;
+
+    FineDoFHandlerViewCell
+    get_cell(const typename DoFHandler<dim>::cell_iterator &cell) const override
+    {
+      return FineDoFHandlerViewCell(
+        [this, cell]() {
+          if (this->mg_level_fine == numbers::invalid_unsigned_int)
+            {
+              const auto cell_id = cell->id();
+              const auto cell_fine_raw =
+                dof_handler_fine.get_triangulation().create_cell_iterator(
+                  cell_id);
+              return cell_fine_raw->has_children();
+            }
+          else
+            {
+              return cell->has_children();
+            }
+        },
+        [this, cell](auto &dof_indices) {
+          if (this->mg_level_fine == numbers::invalid_unsigned_int)
+            {
+              const auto cell_id = cell->id();
+              const auto cell_fine_raw =
+                dof_handler_fine.get_triangulation().create_cell_iterator(
+                  cell_id);
+              return cell_fine_raw->as_dof_handler_iterator(dof_handler_fine)
+                ->get_dof_indices(dof_indices);
+            }
+          else
+            {
+              cell->get_mg_dof_indices(dof_indices);
+            }
+        },
+        [this, cell]() {
+          if (this->mg_level_fine == numbers::invalid_unsigned_int)
+            {
+              const auto cell_id = cell->id();
+              const auto cell_fine_raw =
+                dof_handler_fine.get_triangulation().create_cell_iterator(
+                  cell_id);
```

Comment:
Can you explain why we need to take the detour over `cell->id()` before we use `as_dof_handler_iterator`? Can the underlying triangulations be different and is that a usual case?